src/commands/verify.ts

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+import { CliVerifyArguments } from '../utils/define-args';
+import {
+    generateReport,
+    ValidationResult,
+    VerifyError,
+} from '../utils/generate-report';
+import { buildUrl } from '../utils/build-url';
+import { VALIDATION_ERROR_MESSAGE } from '../utils/errors';
+
+export async function verify(argv: CliVerifyArguments) {
+    const {
+        serviceName,
+        url,
+        serviceVersion,
+        environment,
+        outFile,
+        outDir,
+    } = argv;
+
+    const judgeDUrl = buildUrl(url, serviceName, serviceVersion, environment);
+
+    let validationResults: ValidationResult[] | undefined;
+    let verifyError: VerifyError | undefined;
+
+    try {
+        const response = await axios.get<ValidationResult[]>(judgeDUrl);
+        validationResults = response.data;
+
+        const hasAnyInteractionFailed = response.data.some((result) =>
+            result.interactions.some(
+                (interaction) => interaction.validationResult === 'FAILED'
+            )
+        );
+
+        if (hasAnyInteractionFailed) {
+            throw new Error(VALIDATION_ERROR_MESSAGE);
+        }
+    } catch (error) {
+        if (error.isAxiosError) {
+            verifyError = {
+                message: error.message,
+                status: error.response?.status,
+                path: error.config.url,
+            };
+        }
+        throw error;
+    } finally {
+        await generateReport(outDir, outFile, validationResults, verifyError);
```

Comment:
We should move the invocation of `generateReport` to `index.ts`, so that `verify` has single responsibility and no side-effects apart from http request